core/src/main/java/com/scalar/db/storage/dynamo/QueryScanner.java

```diff
@@ -0,0 +1,100 @@
+package com.scalar.db.storage.dynamo;
+
+import com.scalar.db.api.Result;
+import com.scalar.db.api.Scanner;
+import com.scalar.db.storage.common.ScannerIterator;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+import software.amazon.awssdk.services.dynamodb.model.QueryRequest;
+import software.amazon.awssdk.services.dynamodb.model.QueryResponse;
+
+public class QueryScanner implements Scanner {
+
+  private final DynamoDbClient client;
+  private final QueryRequest request;
+  private final ResultInterpreter resultInterpreter;
+
+  private Iterator<Map<String, AttributeValue>> itemsIterator;
+  @Nullable private Map<String, AttributeValue> lastEvaluatedKey;
+  private int totalResultCount;
+
+  private ScannerIterator scannerIterator;
+
+  public QueryScanner(
+      DynamoDbClient client, QueryRequest request, ResultInterpreter resultInterpreter) {
+    this.client = client;
+    this.request = request;
+    this.resultInterpreter = resultInterpreter;
+
+    query(request);
+  }
+
+  @Override
+  @Nonnull
+  public Optional<Result> one() {
+    if (!hasNext()) {
+      return Optional.empty();
+    }
+
+    return Optional.of(resultInterpreter.interpret(itemsIterator.next()));
+  }
+
+  private boolean hasNext() {
+    if (itemsIterator.hasNext()) {
+      return true;
+    }
+    if (lastEvaluatedKey != null) {
+      QueryRequest.Builder builder = request.toBuilder();
+      builder.exclusiveStartKey(lastEvaluatedKey);
```

Comment:
If the query response has `lastEvaluatedKey`, we need to re-query with `exclusiveStartKey()`.